src/osm_configurator/model/project/configuration/category.py

```diff
@@ -145,85 +141,72 @@ def set_category_name(self, new_category_name):
             return True
         return False
 
-    def get_calculate_area(self):
+    def get_activated_attribute(self) -> List[Attribute]:
         """
-        This says if the area of the category should be calculated or not.
+        Return a list of all used attributes, of the categories.
+        This is used to know which tags we need to save.
 
         Returns:
-            bool:  true if it should get calculated, false if not.
+            List[Attribute]: A list that contains all used attributes
         """
-        pass
+        _activated = []
 
-    def get_calculation_method_of_area(self):
-        """
-        Getter for the calculated area method.
+        for enum_name in self._attributes:
+            if self._attributes.get(enum_name):
+                _activated.append(enum_name)
+        return _activated
 
-        Returns:
-             calculation_method_of_area_enum.CalculationMethodOfArea: The method with which we calculate the area.
+    def get_attribute(self, attribute) -> bool:
         """
-        return self._calculate_area
-
-    def set_calculation_method_of_area(self, new_calculate_area):
-        """
-        Overwrites current calculate_area with the given value.
+        Returns if a given attribute is activated or not.
 
         Args:
-            new_calculate_area (bool): new value that will overwrite the existing value.
-        """
-        self._calculate_area = new_calculate_area
-
-    def get_calculate_floor_area(self):
-        """
-        This says if  the floor area should be calculated or not.
+            attribute (Attribute): The attribute which is inspected.
 
         Returns:
-             bool: true if the floor are should be calculated, otherwise false.
+            bool: True when the attribute is active, otherwise false.
         """
-        return self._calculate_floor_area
+        if self._attributes.get(attribute.name):
+            return True
+        else:
+            return False
 
-    def set_calculate_floor_area(self, new_calculate_floor_area):
+    def set_attribute(self, attribute, boolean):
         """
-        Overwrites the existing instance of calculate_floor_area.
+        Activates and deactivates a given attribute .
 
         Args:
-            new_calculate_floor_area (bool): new value for calculate_floor_are.
+            attribute (Attribute): The attribute which should be activated or deactivated.
+            boolean (bool): The new value.
 
         Returns:
-            bool: True, if the overwriting process was successful, else false.
+            bool: True when it works, otherwise false.
         """
-        if self._calculate_floor_area != new_calculate_floor_area:
-            self._calculate_floor_area = new_calculate_floor_area
-            return True
-        return False
+        self._attributes[attribute.name] = boolean
+        return True
 
-    def get_strictly_use_default_values(self):
+    def get_calculation_method_of_area(self):
```

Comment:
typehint

Reply:
this is from the config branch, it must be fixed there or it will create conflicts